lib/awilixManager.ts

```diff
@@ -8,15 +8,21 @@ declare module 'awilix' {
     asyncDispose?: boolean | string | ((instance: T) => Promise<unknown>)
     asyncDisposePriority?: number // lower means it gets disposed earlier
     eagerInject?: boolean | string
+    tags?: string[]
     enabled?: boolean
   }
 }
 
+interface AssociativeArray {
+  [key: string]: object
+}
+
 export type AwilixManagerConfig = {
   diContainer: AwilixContainer
   asyncInit?: boolean
   asyncDispose?: boolean
   eagerInject?: boolean
+  tags?: string[]
```

Comment:
what is this flag used for?

Reply:
This is the option where the user can define the tags for their dependency, e.g.

```
diContainer.register(
        'dependency2',
        asClass(AsyncInitGetClass, {
          lifetime: 'SINGLETON',
          asyncInit: true,
          asyncInitPriority: 1,
          tags: ['engine', 'ChatGPT-4'] <<< HERE
        }),
      )
```

Without this we get a type error.

Reply:
there are two configs, one for manager, one for dependency. this one is for manager and I don't think we use it